File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//Functions Declarations
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

//Global Variables Declarations
float letters = 0;
float words = 1;
float sentences = 0;

//Main Function make the math and print the result
int main(void)
{
    //Ask for the original Text
    string text = get_string("Text: ");

    //Prints back the text
    //printf("%s\n ", text);

    //Call the functions
    count_letters(text);
    count_words(text);
    count_sentences(text);

    //Make the Math for the Coleman-Liau index with the formula index = 0.0588 * L - 0.296 * S - 15.8
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    //printf("%f L\n %f S\n", L, S);
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    //Number of letters, words and sentences
    //printf("%1ld letters\n %1ld words\n %1ld sentences\n", lroundf(letters), lroundf(words), lroundf(sentences));

    //Prints the results
    if (index > 16)
    {
        printf("Grade 16+\n");
    }

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    if ((index > 1) && (index < 16))
    {
        printf("Grade %ld\n", lroundf(index));
    }


}

//Function that count the letters
int count_letters(string text)
{
    for (int j = 0; j <= 25; j++)
    {
        for (int i = 0; i < strlen(text); i++)
        {
            if (text[i] == 65 + j || text[i] == 97 + j)
            {
                letters++;
            }

        }
    }

    return letters;

}

//Function that count the words
int count_words(string text)
{

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 32)
        {
            words++;
        }

    }

    return words;
}

//Function that count the sentenses
int count_sentences(string text)
{

    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == 33) || (text[i] == 46) || (text[i] == 63))
        {
            sentences++;
        }

    }

    return sentences;
}

File: sentimental-readability/readability.py
# TODO
import cs50


# Main Function make the math and print the result
def main():

    # Ask for the original Text
    text = cs50.get_string("Text: ")

    # Call the functions
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Make the Math for the Coleman-Liau index with the formula index = 0.0588 * L - 0.296 * S - 15.8
    # print(letters, words, sentences)
    L = (letters / words) * 100
    S = (sentences / words) * 100
    # print(L, S)
    index = (0.0588 * L) - (0.296 * S) - 15.8

    # Prints the results
    if (index > 16):
        print("Grade 16+")

    if (index < 1):
        print("Before Grade 1")

    if ((index > 1) and (index < 16)):
        print("Grade", round(index))


# Function that count the letters
def count_letters(text):
    letters = 0
    for j in range(27):
        for i in range(len(text)):
            if (text[i] == chr(65 + j) or text[i] == chr(97 + j)):
                letters += 1
    return letters


# Function that count the words
def count_words(text):
    words = 1
    for i in range(len(text)):
        if (text[i] == chr(32)):
            words += 1
    return words


# Function that count the sentenses
def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if ((text[i] == chr(33)) or (text[i] == chr(46)) or (text[i] == chr(63))):
            sentences += 1
    return sentences


if __name__ == "__main__":
    main()